### PR TITLE
Fetch test data from GIN

### PR DESCRIPTION
**Pull request by niksirbi**
## Aim

Till now test data were stored within the repository, inside a ``data`` folder.
As more features will be added, the amount of test data will grow. Moreover, we will also need data for examples/tutorials.
To avoid storing large data files on GitHub, I have created a [data repository on GIN](https://gin.g-node.org/neuroinformatics/movement-test-data). This PR adds code to fetch the data from GIN using [pooch](https://www.fatiando.org/pooch/latest/).

## Implementation
I decided to create a proper `datasets.py` module, to provide a unified interface for fetching all kinds of data - for testing, tutorials, or otherwise.

Currently, it contains a single function - `fetch_pose_data.py()` for fetching sample pose estimation data. I think this approach is generalisable since more functions can be added for other kinds of data - e.g. models, and labels. Instead of grouping the data via data type, we could also think of grouping them according to purpose and naming the fetch functions accordingly (e.g. `fetch_test_data()`, `fetch_tutorial_data()`). I haven't done this here, but would be easy to switch to this model if the need arises.

## Review needs
I've decided to tag you @adamltyson as a reviewer for this one. The rationale is that if this implementation is solid, I will replicate it across other projects as well, starting with WAZP and NeuralPlayground. So it's important to get this right the first time.

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Review by adamltyson — APPROVED** [review]
Looks good to me, I left one small comment. 

I was thinking it would be nice to abstract away the filenames, so something like `sample_wasp_poses()` would work, but I can't think of an elegant way to do it.

File: movement/datasets.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        "DLC_single-wasp.predictions.h5": "931dddb6ef5e08db6054d3757a441ee31dd0d9ff5a10802ad8405d6c4e7e274e",  # noqa: E501
+        "DLC_single-wasp.predictions.csv": "9b194cc930c2e2e0d33c816320d029f889b306d53ff2fe95ff408e99c9cdea23",  # noqa: E501
+        "DLC_single-mouse_EPM.predictions.h5": "0ddc2b08c9401435929783b22ea31b3673ceb80c3a02c5f3531bb1cfd78deea5",  # noqa: E501
+        "SLEAP_single-mouse_EPM.predictions.slp": "6ede1837dc8a66e5615ff1dc8d2d590727fa78e9b4d9391bad9e9ef0922445fe",  # noqa: E501
+        "SLEAP_two-mice_social-interaction.predictions.slp": "45881affde9704c045e70b8d4b3f6bbb8d9bd8ef9f4cdea6d173cfe35857549b",  # noqa: E501
+    },
+)
+
+
+def fetch_pose_data(filename: str) -> Path:
```
[the remark this answers is not in the document]
Comment (niksirbi) [reply movement/datasets.py]: Agreed on this one. Renamed to avoid confusion.

**Comment by niksirbi** [comment]
> I was thinking it would be nice to abstract away the filenames, so something like `sample_wasp_poses()` would work, but I can't think of an elegant way to do it.

I like this idea, especially for sample data used in tutorials etc. However, I'll open an issue and leave this for later. When we have more data files, it will become clearer which abstractions are useful and how to name them.

**Comment by adamltyson** [comment]
> I like this idea, especially for sample data used in tutorials etc. However, I'll open an issue and leave this for later. When we have more data files, it will become clearer which abstractions are useful and how to name them.

Sounds good, feel free to merge.

**Pull request merged**